sklearn/datasets/_twenty_newsgroups.py

```diff
@@ -433,12 +451,22 @@ def fetch_20newsgroups_vectorized(*, subset="train", remove=(), data_home=None,
                                    download_if_missing=download_if_missing)
 
     if os.path.exists(target_file):
-        X_train, X_test = joblib.load(target_file)
+        try:
+            X_train, X_test, feature_names = joblib.load(target_file)
+        except ValueError as e:
+            raise ValueError(
+                f"The cached dataset located in {target_file} was fetched "
+                f"with an older scikit-learn version and it is not compatible "
+                f"with the scikit-learn version imported. You need to "
```

Comment:
Technically, we don't need f-strings for those 2 lines.

Reply:
yeah, but some of us prefer to have the `f` at the beginning of each line cause it looks better :P